src/lsm/binary_search.zig

```diff
@@ -18,11 +23,23 @@ pub fn binary_search_values_raw(
     values: []const Value,
     key: Key,
 ) u32 {
-    assert(values.len > 0);
+    if (values.len == 0) return 0;
+
+    if (config.verify) {
+        // Input must be sorted by key.
+        for (values) |_, i| {
+            assert(i == 0 or compare_keys(key_from_value(&values[i - 1]), key_from_value(&values[i])) != .gt);
```

Comment:
Codebase style is to wrap at 100 chars. Not sure what `zig fmt` will allow, but something like
```suggestion
            assert(i == 0 or
                compare_keys(key_from_value(&values[i - 1]), key_from_value(&values[i])) != .gt);
```

Reply:
Should the lint be running in CI?

```
zig run ./scripts/lint.zig -- $(rg --files src/ | grep '.zig') 2>&1 | grep -c 'line exceeds 100'
258
```

Reply:
Not for now. Fixing all of those would cause a bunch of git conflicts with open branches right now. Maybe we could try to fix it gradually, and then when the number is much lower add it to CI.